File: python-playwright-enterprise-framework/utils/logger.py
import logging
from pathlib import Path


ROOT_DIR = Path(__file__).resolve().parents[1]
LOG_DIR = ROOT_DIR / "reports" / "logs"
LOG_FILE = LOG_DIR / "test_execution.log"


def get_logger(name=__name__):
    LOG_DIR.mkdir(parents=True, exist_ok=True)

    logger = logging.getLogger(name)
    logger.setLevel(logging.DEBUG)

    if logger.handlers:
        return logger

    formatter = logging.Formatter(
        "%(asctime)s | %(levelname)s | %(name)s | %(message)s"
    )

    file_handler = logging.FileHandler(LOG_FILE, encoding="utf-8")
    file_handler.setLevel(logging.INFO)
    file_handler.setFormatter(formatter)

    console_handler = logging.StreamHandler()
    console_handler.setLevel(logging.INFO)
    console_handler.setFormatter(formatter)

    logger.addHandler(file_handler)
    logger.addHandler(console_handler)

    return logger

File: javascript-playwright-enterprise-framework/utils/logger.js
import fs from "node:fs";
import path from "node:path";
import { ROOT_DIR } from "./config_reader.js";

const LOG_DIR = path.join(ROOT_DIR, "reports", "logs");
const LOG_FILE = path.join(LOG_DIR, "test_execution.log");

function formatMessage(message, args) {
  return args.reduce((formatted, value) => formatted.replace("%s", String(value)), message);
}

export function getLogger(name = "default") {
  fs.mkdirSync(LOG_DIR, { recursive: true });

  function write(level, message, args = []) {
    const line = `${new Date().toISOString()} | ${level} | ${name} | ${formatMessage(message, args)}\n`;
    fs.appendFileSync(LOG_FILE, line, "utf-8");
    process.stdout.write(line);
  }

  return {
    info: (message, ...args) => write("INFO", message, args),
    warn: (message, ...args) => write("WARN", message, args),
    error: (message, ...args) => write("ERROR", message, args),
    debug: (message, ...args) => write("DEBUG", message, args)
  };
}
